app/controllers/admin/assets_controller.rb

```diff
@@ -23,6 +23,8 @@ def index
   def show
     @asset = Asset.find_by_friendlier_id!(params[:id])
     authorize! :read, @asset
+
+    @edit_asset_path = @asset.collection_thumbnail? ? edit_admin_collection_path(@asset.parent) : edit_admin_asset_path(@asset)
```

Comment:
can we make this a controller method instead, that is made available to views with `helper_method :name_of_controller_method`?  

instance variables to communicate state between controllers and views... while Rails kind of suggests it... experience of many shows that they are just messy and hard to keep track of and easy to mess up.